tools/update_unicode_iso8859_tables.dart

```diff
@@ -0,0 +1,300 @@
+///
+/// Ends region started by [generatedContentLead].
+const generatedContentTail = '// -- END GENERATED CONTENT --';
+
+// Rebuild the tables used for ISO-8859 code pages.
+void main(List<String> args) async {
+  var tableFile = File(tableFilePath);
+  if (!tableFile.existsSync()) {
+    stderr.writeln('Cannot find file to generate into: $tableFile');
+    stderr.writeln('Create file with proper copyright header and run again.');
+  }
+  var tableFileContent = tableFile.readAsStringSync();
+
+  // Generate new declarations.
+  var declarations = await generateTables();
+
+  // Find generated content markers in existing file.
+  var generatedContentMatch = RegExp('$generatedContentLead'
+          r'.*\r?\n(?:// Generated.*\r?\n)?([^]*?\r?\n)'
+          '$generatedContentTail'
+          r'(.*\r?\n)?'
+          r'|$')
+      .firstMatch(tableFileContent)!;
+
+  var existingDeclarations = generatedContentMatch[1] ?? '';
+  if (declarations == existingDeclarations) {
+    // If no change, no write.
+    stderr.writeln('No changes');
+    return;
+  }
+  var replacement = replacementString(declarations);
+  // Guess file's newline, change to '\r\n' if necessary.
+  if (Platform.isWindows && tableFileContent.contains('\r\n')) {
+    replacement = replacement.replaceAll('\n', '\r\n');
+  }
+  tableFileContent = tableFileContent.replaceRange(
+      generatedContentMatch.start, generatedContentMatch.end, replacement);
+  tableFile.writeAsStringSync(tableFileContent);
+  print('Written: $tableFile');
+}
+
+/// Shared HTTP client used by [fetchTable], initialized if needed.
+///
+/// Closed by [generateTables] when done fetching.
+HttpClient? httpClient;
+
+/// Fetches table file content, parses and builds constant string declarations.
+Future<String> generateTables() async {
+  // Used for caching fetched files.
+  // Useful if running the script multiple times.
+  var cacheDir = Directory(cachePath);
+  if (!cacheDir.existsSync()) cacheDir.createSync(recursive: true);
+
+  var buffer = StringBuffer();
+  // For each ISO-8859-n, n > 1, fetch the table,
+  // and generate a constant string corresponding to the mapping
+  // into `buffer`.
+  // There is no ISO-8859-12. Don't include ISO-8859-1,
+  // since it's already included in `dart:convert`.
+  for (var i = 2; i <= 16; i++) {
+    if (i == 12) continue; // No ISO-8859-12.
+    await generateTableFor(buffer, i);
+  }
+
+  httpClient?.close();
+  return buffer.toString();
+}
+
+/// Generates the constant string for a single ISO-8859 code page.
+Future<void> generateTableFor(StringSink buffer, int isoNumber) async {
+  var (uri, content) = await fetchTable(isoNumber);
+  var text = parseTable(isoNumber, content);
+  buffer
+    ..writeln()
+    ..writeln('// Characters of ISO-8859-$isoNumber.')
+    ..write('const iso8859_$isoNumber = // From $uri');
+  // Write 8 chars per line, as Unicode escapes.
+  for (var i = 0; i < text.length;) {
+    var charsOnLine = nextRangeLength(text, i, 74);
```

Comment:
Would this be more readable if `nextRangeLength` were a generator? So you could write:

```dart
  for(var next in splitTextIntoLines(text, 74)) {
    ...
  }
```

Reply:
Could work too.
And could let me ensure that every line is size-aligned, so a 16-character line will always start at position that is a multiplum of 16. (Which is what I really want, and I think I actually got it entirely by happy accident.)